classes/context/Context.php

```diff
@@ -546,6 +546,26 @@ public function isInstitutionStatsEnabled(Site $site): bool
         }
         return (bool) $enableInstitutionUsageStats;
     }
+
+    /**
+    * What Geo data to track for usage statistics.
+    * Consider context setting only if the site setting is enabled and context setting considers less Geo data than site setting.
+    *
+    * @return ?string Possible return values: null, disabled, country, coutnry+region, country+region+city
+    */
+    public function getGeoStatsSetting(Site $site): ?string
+    {
+        $enableGeoUsageStats = $site->getData('enableGeoUsageStats');
+        if ($enableGeoUsageStats !== null) {
+            if (($enableGeoUsageStats != 'disabled') &&
+                ($this->getData('enableGeoUsageStats') !== null) &&
+                ($this->getData('enableGeoUsageStats') != $site->getData('enableGeoUsageStats')) &&
+                    str_starts_with($enableGeoUsageStats, $this->getData('enableGeoUsageStats'))) {
+                $enableGeoUsageStats = $this->getData('enableGeoUsageStats');
+            }
+        }
```

Comment:
I admit it took me a while to understand the logic here. What about being a bit more explicit about the different states that can happen?

```php
$siteSetting = $site->getData('enableGeoUsageStats');
if ($siteSetting !== null || $siteSetting === 'disabled') {
    return $siteSetting;
}
$contextSetting = $this->getData('enableGeoUsageStats');
if (str_starts_with($siteSetting, $contextSetting)) {
    return $contextSetting;
}
return $siteSetting;
```